TF_Tooling_Web/src/Plugins/tests/InputActionManager.test.ts

```diff
@@ -0,0 +1,64 @@
+import { HandChirality, HandType, InputType, InteractionType, TouchFreeInputAction } from '../../TouchFreeToolingTypes';
+import { mockTfPluginInputAction } from '../../tests/testUtils';
+import { InputActionManager } from '../InputActionManager';
+import { InputActionPlugin } from '../InputActionPlugin';
+
+describe('InputActionManager', () => {
+    let currentInputAction: TouchFreeInputAction;
+    let currentModifiedAction: TouchFreeInputAction;
+    let pluginCalled = false;
+
+    class MockPlugin extends InputActionPlugin {
+        RunPlugin(inputAction: TouchFreeInputAction): TouchFreeInputAction | null {
+            pluginCalled = true;
+            expect(inputAction).toStrictEqual(currentInputAction);
+            const modifiedInputAction = super.RunPlugin(inputAction);
+            expect(modifiedInputAction).toStrictEqual(currentModifiedAction);
+            return modifiedInputAction;
+        }
+
+        ModifyInputAction(inputAction: TouchFreeInputAction): TouchFreeInputAction | null {
+            currentModifiedAction = { ...inputAction, Timestamp: Date.now() };
+            return currentModifiedAction;
+        }
+
+        TransmitInputAction(inputAction: TouchFreeInputAction): void {
+            expect(inputAction).toStrictEqual(currentModifiedAction);
+        }
+    }
+
+    beforeAll(() => {
+        InputActionManager.SetPlugins([new MockPlugin()]);
+    });
+
+    test('Check plugin gets called with the correct data', () => {
+        expect(pluginCalled).toBeFalsy();
+        currentInputAction = new TouchFreeInputAction(
+            Date.now(),
```

Comment:
:eyes: As per other comment, this could use that same function.

Reply:
I might be misunderstanding, but using your new `createInputAction` function here instead of the `new TouchFreeInputAction` only reduces code duplication and doesn't do any object assignment or destructuring?

Reply:
Ah glossed over these were all the default parameters so can just call `createInputAction` without passing it anything

Reply:
Updated